src/machine/operation.c

```diff
@@ -154,3 +154,19 @@ Operation *operation_free(Operation *o) {
 
         return mfree(o);
 }
+
+_noreturn_ void report_errno_and_exit(int errno_fd, int r) {
+        assert(errno_fd >= 0);
+
+        if (r < 0) {
```

Comment:
maybe invert this, in the interest of shallow indentation. 

we typically prefer early exit over deep indentation